src/Plugin/Notifier/Email.php

```diff
@@ -71,6 +72,13 @@ public static function create(ContainerInterface $container, array $configuratio
   public function deliver(array $output = []) {
     /** @var \Drupal\user\UserInterface $account */
     $account = $this->message->uid->entity;
+
+    if (!$this->configuration['mail'] && !$account->id()) {
```

Comment:
We don't need to check for `mail` here, since if it isn't passed, it gets set below from the account object.